messaging/WFLY-14133_configurable_bridge_call_timeout.adoc

```diff
@@ -0,0 +1,129 @@
+
+The Artemis broker defines a call-timeout attribute on the core bridge, which sets a time out for blocking calls to the
+server. Unless modified, the call-timeout defaults to 30 seconds.
+
+It was requested by GSS to make this attribute configurable via the Wildfly/JBoss EAP management model.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.redhat.com/browse/WFLY-14133
+
+=== Related Issues
+
+* https://issues.apache.org/jira/browse/ARTEMIS-3074
+* https://issues.redhat.com/browse/JBEAP-20326
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+=== Testing By
+// Put an x in the relevant field to indicate if testing will be done by Engineering or QE.
+// Discuss with QE during the Kickoff state to decide this
+[X] Engineering
+
+[ ] QE
+
+=== Affected Projects or Components
+
+* Wildfly
+* ActiveMQ Artemis
+
+=== Other Interested Projects
+
+== Requirements
+
+=== Hard Requirements
+
+* Make the `BridgeConfiguration#callTimeout` attribute configurable via the Wildfly management model.
+
+=== Nice-to-Have Requirements
+
+None.
+
+=== Non-Requirements
+
+None.
+
+== Implementation Plan
+
+Relevant entities:
+
+* `ActiveMQServerControl` is a part of the high level Artemis API used to control the embedded Artemis broker.
+* `ActiveMQServer` is part of a lower level Artemis API used to control the embedded Artemis broker.
+* `BridgeDefinition` is a Wildfly management model resource definition class describing the Artemis core bridge.
+* `BridgeAdd` is a Wildfly controller handler used to create the Artemis core bridges.
+
+Code changes:
+
+1. Add the `call-timeout` attribute to the `BridgeDefinition` class in the messaging-activemq subsystem.
+Depending on how this work aligns with other management model changes, this may require creating new schema version
+of the messaging management model.
+
+2. Modify the `BridgeAdd` class so that the `BridgeDefinition#call-timeout` management model value is propagated to the
+code responsible for creating the bridge. Currently, the attribute values are passed
+to the `ActiveMQServerControl#createBridge()` method (part of Artemis API). There are several variants of this method,
+none of these however accepts the `call-timeout` attribute. There are two options how to overcome this:
+
+** Add another variant of the `ActiveMQServerControl#createBridge()` method which would accept the `callTimeout` attribute.
+This is the preferred option but requires a change in the Artemis codebase.
+After discussion with AMQ team, the new variant of the `createBridge()` method would accept a single String parameter.
+The passed string would contain a JSON encoded representation of the bridge configuration.
+
+** Instead of calling `ActiveMQServerControl#createBridge()`, the bridge can be created with a lower level API
+by calling `ActiveMQServer#deployBridge()`. This has been discussed with
+the AMQ team as a viable option, but requires more code on the Wildfly side, some of which would be duplicating
```

Comment:
> but requires more code on the Wildfly side

AFAIU [current implementation uses deployBridge](https://github.com/wildfly/wildfly/pull/13858/files#diff-a04d5dc35c0c963347d1592e8a551b95d13e87e118d743c171a8b3d455227705R183). What do you mean by "more code" here?

Reply:
This PR will need to be reworked.

Reply:
To answer this, couple of methods like following would have to be replicated in Wildfly code:

* ActiveMQServerControlImpl#checkStarted()
* ActiveMQServerControlImpl#clearIO()
* ActiveMQServerControlImpl#blockOnIO()